### PR TITLE
chat gpt for elegance on Will's functions and carrying the output file input through

### PR DESCRIPTION
**Pull request by wcornwell**
Check out how good Chat-GPT is at decomposing functions.  Worked the first time.

### CODE REVIEW

**Comment by fontikar** [comment]
Totally nitpicky, do we want to allow users to save the output when full = FALSE? Lines 528 - 537. 

If so, may need another if statement with the else on line 527

File: R/clean_names.R [hunk 1]
```diff
@@ -502,136 +504,140 @@ standardise_names <- function(taxon_names) {
...
-  } else {
-    return(TRUE)
   }
-}
 
-
-
-#not working yet
+#' @noRd
 find_mrct <- function(taxa,
```
Comment (dfalster) [inline R/clean_names.R:609]: What does mrct stand for?

File: R/state_diversity.R [hunk 2]
```diff
@@ -23,87 +23,69 @@
...
-    
-    #go through the states one by one
-    for (i in 1:length(apc_places)) {
-      species_df <-
-        state_parse_and_add_column(species_df, apc_places[i], apc_species)
-    }
-    return(species_df)
+#'
+#'
+create_species_state_origin_matrix <- function(resources = load_taxonomic_resources()) {
```
Comment (dfalster) [inline R/state_diversity.R:28]: As nice as it looks, I don't think this code will run and produce the same outputs as previously.

**Review by dfalster — DISMISSED** [review]
Before accepting a refactoring like this, we need some more tests to check specific outputs. Most/all of our existing tests just check that there was output created, with easy-to-pass tests like n >100

I'm unwilling to approve until we can show it produces the same results. 

I'm very sceptical the new code for state-diversity will produce the same results
In any case, we don't know as we don't have any tests that would tell us. the current tests are very lenient,int

**Review by fontikar — DISMISSED** [review]
> most_likely_species defaults to the original_name if that name is accepted by the APC; this will be right for certain species subsets, but make errors in other cases, use with caution. 

That worries me a little as an argument. Let's keep for now, but we should come up with use cases for each of these methods for documentation and clever code to avoid errors. If its too hard maybe collpase to most_likely_species is too risky as an option!

**Comment by dfalster** [comment]
> Before accepting a refactoring like this, we need some more tests to check specific outputs. Most/all of our existing tests just check that there was output created, with easy-to-pass tests like n >100
> 
> I'm unwilling to approve until we can show it produces the same results.
> 
> I'm very sceptical the new code for state-diversity will produce the same results In any case, we don't know as we don't have any tests that would tell us. the current tests are very lenient,int

Regarding workflow, these tests should be added on a seperate branch off master, then be merged in here

File: R/state_diversity.R (hunk 2 above)
Comment (fontikar) [reply R/state_diversity.R:28]: Let me test it out with the vignette :)

Comment (fontikar) [reply R/state_diversity.R:28]: Seems to be able to render the vignette okay! But that is bare minimum. Did ChatGPT really write all these sub functions???

File: R/clean_names.R (hunk 1 above)
Comment (wcornwell) [reply R/clean_names.R:604]: most recent common taxon

Comment (fontikar) [reply R/clean_names.R:604]: If not working yet, I suggest removing and adding it as a seperate issue/feature as it is not contributing to "carrying output file through functions"

File: R/state_diversity.R (hunk 2 above)
Comment (fontikar) [reply R/state_diversity.R:28]: @wcornwell lets smash out some tests for this this afternoon so we can address some of these concerns!

Comment (wcornwell) [reply R/state_diversity.R:28]: added tests and pulled across

File: R/clean_names.R (hunk 1 above)
Comment (wcornwell) [reply R/clean_names.R:604]: it is working just forgot to remove comment

**Comment by fontikar** [comment]
I forced rerun of jobs, once they pass I am happy for this to merge in :)

**Pull request merged**